app/src/main/java/com/example/places/MapFragment.kt

```diff
@@ -37,7 +39,30 @@ class MapFragment : Fragment(R.layout.map_fragment), OnMapReadyCallback {
     override fun onMapReady(p0: GoogleMap) {
         p0.let {
             googleMap = it
+            viewModel.placesLiveData.observe(this) { places ->
+                markerPlaces(places)
+            }
+            viewModel.selectedPlace.observe(this) { place ->
+                zoomPlace(place)
+            }
+        }
+    }
+
+    fun markerPlaces(placesList: List<Place>) {
+        for (places in placesList) {
+            if (places.latitude != null && places.longitude != null) {
+                val location = LatLng(places.latitude.toDouble(), places.longitude.toDouble())
```

Comment:
This code is the same as the one in "markerPlaces" for zooming. You should call this function in the markerPlaces so you don't have boilerplate code